### PR TITLE
Category for all

### PR DESCRIPTION
**Pull request by MarioGafeed**

### CODE REVIEW

File: app/DataTables/CategoriesDataTable.php [hunk 1]
```diff
@@ -46,7 +47,7 @@ public function dataTable($query)
      */
     public function query()
     {
-        $query = Vcategory::query()->select('vcategories.*');
+        $query = category::query()->select('categories.*');
```
Comment (mo7zayed) [inline app/DataTables/CategoriesDataTable.php:50]: Model should be Category not category

File: app/Http/Controllers/CategoryController.php [hunk 2]
```diff
@@ -1,30 +1,29 @@
...
+use App\DataTables\CategoriesDataTable;
 use App\Authorizable;
 
-class PcategoryController extends Controller
+class CategoryController extends Controller
 {
   // use Authorizable;
-  private $PcategoryInterface;
-  public function __construct(PcategoryInterface $PcategoryInterface)
+  private $CategoryInterface;
```
Comment (mo7zayed) [inline app/Http/Controllers/CategoryController.php:13]: It should be categoryInterface

File: app/Http/Repositories/CategoryRepository.php [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+use Illuminate\Http\Request;
+
+
+
+class CategoryRepository implements CategoryInterface
+{
+    private $viewPath = 'backend.categories';
+    use CategoryTrait;
+    private $categoryModel;
+    public function __construct(Category $cat)
```
Comment (mo7zayed) [inline app/Http/Repositories/CategoryRepository.php:17]: $category not $cat,
Keep it direct

File: config/app.php [hunk 4]
```diff
@@ -175,7 +175,9 @@
         // App\Providers\BroadcastServiceProvider::class,
         App\Providers\EventServiceProvider::class,
         App\Providers\RouteServiceProvider::class,
-        // Yajra\DataTables\DataTablesServiceProvider::class,
+        Spatie\Tags\TagsServiceProvider::class,
+        Yajra\DataTables\DataTablesServiceProvider::class,
+        Spatie\Tags\TagsServiceProvider::class,
```
Comment (mo7zayed) [inline config/app.php:180]: It's duplicate

**Review by mo7zayed — CHANGES_REQUESTED** [review]

File: app/DataTables/CategoriesDataTable.php (hunk 1 above)
Comment (MarioGafeed) [reply app/DataTables/CategoriesDataTable.php:50]: Done

File: app/Http/Controllers/CategoryController.php (hunk 2 above)
Comment (MarioGafeed) [reply app/Http/Controllers/CategoryController.php:13]: Done For All Interfaces

File: config/app.php (hunk 4 above)
Comment (MarioGafeed) [reply config/app.php:180]: Done

File: app/Http/Repositories/CategoryRepository.php (hunk 3 above)
Comment (MarioGafeed) [reply app/Http/Repositories/CategoryRepository.php:17]: بتلخبطني ساعات 
فممكن نخليها كدة مؤقتا..

**Pull request merged**